### PR TITLE
Logical commands

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This PR adds the logical commands. By now, only the stubs are present.
Is there anything missing here? Please tell me.

@waysome/hackers

### CODE REVIEW

**Comment by neithernut** [comment]
Sufficient for now.

**Comment by matthiasbeyer** [comment]
I'm ready for review here.

**Comment by neithernut** [comment]
You forgot to deinit the target value in case it's not an bool.
Also: I think it would be better if the commands were shorter in name, e.g. `lnot`, `land`, `lnand`

**Comment by matthiasbeyer** [comment]
Hm. I don't see any benefits from shorter names and I don't see any drawbacks by the longer ones.

**Comment by neithernut** [comment]
Users have to **type** them.

**Comment by matthiasbeyer** [comment]
Ah, did not thought about that they get exported directly. Okay, ... But I guess users will _not_ have to type them as there will be libraries. I don't think anyone will write a transaction by hand.

But kay, I will fix this.

File: src/command/logical.c [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        }
+
+        ws_value_deinit((struct ws_value*) base);
+        ws_value_bool_init((struct ws_value_bool*) base);
+        ws_value_bool_set((struct ws_value_bool*) base, false);
+        return 0;
+    }
+
+    return 0;
+}
```
Comment (neithernut) [inline src/command/logical.c:70]: This function fails if all values are falsy.

File: src/command/logical.c [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        if (ws_value_is_true(&args->value)) {
+            ws_value_deinit((struct ws_value*) base);
+            ws_value_bool_init((struct ws_value_bool*) base);
+            ws_value_bool_set(&base->bool_, true);
+            return 0;
+        }
+        args++;
+    }
+
+    ws_value_bool_set(&base->bool_, false);
```
Comment (neithernut) [inline src/command/logical.c:106]: You did forget the re-initialisation here.

File: src/command/logical.c [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                return 0;
+            }
+
+            have_true = true;
+        }
+
+        args++;
+    }
+
+    ws_value_bool_set(&args->bool_, have_true);
```
Comment (neithernut) [inline src/command/logical.c]: You did forget the reinitialization here.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**